File: src/main/java/um/haberes/rest/model/Item.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import um.haberes.rest.kotlin.model.Codigo;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = {
		@UniqueConstraint(columnNames = { "legajoId", "anho", "mes", "codigoId" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@ToString
@NoArgsConstructor
@AllArgsConstructor
public class Item extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = -8758620272361908858L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long itemId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private Integer codigoId;
	private String codigoNombre = "";
	private BigDecimal importe = BigDecimal.ZERO;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "codigoId", insertable = false, updatable = false)
	private Codigo codigo;

	public String legajoKey() {
		return this.legajoId + "." + this.anho + "." + this.mes + "." + this.codigoId;
	}

}

File: src/main/java/um/haberes/rest/kotlin/model/Item.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import um.haberes.rest.model.Auditable
import um.haberes.rest.model.Persona
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["legajoId", "anho", "mes", "codigoId"])])
data class Item(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var itemId: Long? = null,
    var legajoId: Long? = null,
    var anho: Int = 0,
    var mes: Int = 0,
    var codigoId: Int? = null,
    var codigoNombre: String = "",
    var importe: BigDecimal = BigDecimal.ZERO,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "codigoId", insertable = false, updatable = false)
    var codigo: Codigo? = null

) : Auditable() {
    fun legajoKey(): String {
        return this.legajoId.toString() + "." + this.anho + "." + this.mes + "." + this.codigoId
    }

}
